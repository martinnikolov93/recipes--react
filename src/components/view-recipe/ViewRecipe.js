import React from 'react';

import styles from './ViewRecipe.module.css'
import FavouriteIcon from '../favourite-icon/FavouriteIcon';
import { isEmptyObject } from '../../utils/helpers';


const ViewRecipe = (props) => {
    if(isEmptyObject(props.recipe)){
        return 'Loading..'
    }

    return (
        <>
            <article>
                <img className={styles.image} src={props.recipe.url} alt={props.recipe.title} />
                <FavouriteIcon recipeId={props.recipe._id} />
                <h3>{props.recipe.title}</h3>
                <div>{props.recipe.description + "..."}</div>
            </article>
        </>
    )
}

export default ViewRecipe