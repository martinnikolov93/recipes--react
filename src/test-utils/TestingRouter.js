import React from 'react';
import UserContext from '../Context';
import { BrowserRouter, Route } from 'react-router-dom'

const TestingRouter = ({user, children}) => {
    return (
        <BrowserRouter>
            <Route>
                <UserContext.Provider value={user}>
                    {children}
                </UserContext.Provider>
            </Route>
        </BrowserRouter>
    )
}

export default TestingRouter