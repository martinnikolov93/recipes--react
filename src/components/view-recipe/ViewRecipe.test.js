import React from 'react';
import renderer from 'react-test-renderer'
import ViewRecipe from './ViewRecipe';
import TestingRouter from '../../test-utils/TestingRouter';

jest.mock('../favourite-icon/FavouriteIcon', () => 'FavouriteIcon')
jest.mock('../edit-recipe-button/EditRecipeButton', () => 'EditRecipeButton')
jest.mock('../add-review/AddReview', () => 'AddReview')

const dataDummy = {
    title: 'Test title',
    url: 'http://google.com',
    description: 'Test description',
    author: {
        _id: '123'
    }
}

const userDummyCreator = {
    user: {
        _id: '123'
    }
}

const userDummyNotCreator = {
    user: {
        _id: '234'
    }
}

describe('ViewRecipe component', () => {
    it('Should render ViewRecipe component when is recipe creator', () => {
        const tree = renderer.create(
            <TestingRouter user={userDummyCreator}>
                <ViewRecipe data={dataDummy} />
            </TestingRouter>
        ).toJSON()

        expect(tree).toMatchSnapshot()
    })

    it('Should render ViewRecipe component when is NOT recipe creator', () => {
        const tree = renderer.create(
            <TestingRouter user={userDummyNotCreator}>
                <ViewRecipe data={dataDummy} />
            </TestingRouter>
        ).toJSON()

        expect(tree).toMatchSnapshot()
    })
})