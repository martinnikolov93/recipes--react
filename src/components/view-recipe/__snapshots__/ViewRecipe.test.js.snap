// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ViewRecipe component Should render ViewRecipe component when is NOT recipe creator 1`] = `
<article>
  <img
    alt="Test title"
    className="image"
    src="http://google.com"
  />
  <div>
    <FavouriteIcon />
  </div>
  <hr
    className="recipe-hr"
  />
  <h3>
    Test title
  </h3>
  <textarea
    className="description"
    disabled={true}
  >
    Test description
  </textarea>
  <br />
  <br />
  <h2>
    Rate
  </h2>
  <hr
    className="recipe-hr"
  />
  <div>
    <AddReview />
  </div>
</article>
`;

exports[`ViewRecipe component Should render ViewRecipe component when is recipe creator 1`] = `
<article>
  <img
    alt="Test title"
    className="image"
    src="http://google.com"
  />
  <div>
    <FavouriteIcon />
    <EditRecipeButton />
  </div>
  <hr
    className="recipe-hr"
  />
  <h3>
    Test title
  </h3>
  <textarea
    className="description"
    disabled={true}
  >
    Test description
  </textarea>
  <br />
  <br />
  <h2>
    Rate
  </h2>
  <hr
    className="recipe-hr"
  />
  <div>
    <AddReview />
  </div>
</article>
`;
